pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -993,3 +976,35 @@ func (m *kubeGenericRuntimeManager) removeContainerLog(containerID string) error
 func (m *kubeGenericRuntimeManager) DeleteContainer(containerID kubecontainer.ContainerID) error {
 	return m.removeContainer(containerID.ID)
 }
+
+// setTerminationGracePeriod determines the grace period to use when killing a container
+func setTerminationGracePeriod(pod *v1.Pod, containerSpec *v1.Container, reason containerKillReason) int64 {
+	gracePeriod := int64(minimumGracePeriodInSeconds)
+	switch {
+	case pod.DeletionGracePeriodSeconds != nil:
+		return *pod.DeletionGracePeriodSeconds
+	case pod.Spec.TerminationGracePeriodSeconds != nil:
+		switch reason {
+		case reasonStartupProbe:
+			if isProbeTerminationGracePeriodSecondsSet(containerSpec.StartupProbe, "StartupProbe", pod.Spec.TerminationGracePeriodSeconds) {
+				return *containerSpec.StartupProbe.TerminationGracePeriodSeconds
+			}
+		case reasonLivenessProbe:
+			if isProbeTerminationGracePeriodSecondsSet(containerSpec.LivenessProbe, "LivenessProbe", pod.Spec.TerminationGracePeriodSeconds) {
+				return *containerSpec.LivenessProbe.TerminationGracePeriodSeconds
+			}
+		}
+		return *pod.Spec.TerminationGracePeriodSeconds
+	}
+	return gracePeriod
+}
+
+func isProbeTerminationGracePeriodSecondsSet(probe *v1.Probe, probeType string, podPeriod *int64) bool {
+	if probe != nil && probe.TerminationGracePeriodSeconds != nil {
+		if *probe.TerminationGracePeriodSeconds > *podPeriod {
+			klog.V(2).InfoS("Using probe-level grace period that is greater than the pod-level grace period", "probe-type", probeType, "probe-grace-period", probe.TerminationGracePeriodSeconds, "pod-grace-period", podPeriod)
```

Comment:
Thinking this should be level 4, 5, or 6... Maybe 4?

Reply:
+1 on 4, it's vaguely useful workload debugging information.

This log should also include the Pod information and probe name though, otherwise it's ~kinda just noise.

Reply:
updated to mirror the info in [similar log messages](https://github.com/kubernetes/kubernetes/blob/3ae356fb4a1de70c551be53c64130414ea03fb2d/pkg/kubelet/kuberuntime/kuberuntime_container.go#L700-L701)